lib/cog/chat/adapter.ex

```diff
@@ -226,15 +226,29 @@ defmodule Cog.Chat.Adapter do
   # Non-blocking "cast" messages
   def handle_cast(_conn, @adapter_topic, "send",  %{"target" => target,
                                                     "message" => message,
-                                                    "provider" => provider}, state) do
-    case with_provider(provider, state, :send_message, [target, message]) do
+                                                    "provider" => provider,
+                                                    "metadata" => metadata}, state) do
+    metadata = case metadata do
+      map when is_map(map) ->
+        case Cog.Chat.MessageMetadata.from_map(metadata) do
+          {:ok, metadata} ->
+            metadata
+          _error ->
+          %Cog.Chat.MessageMetadata{}
+        end
+      _ ->
+        %Cog.Chat.MessageMetadata{}
```

Comment:
Should log that we're ignoring non-map metadata so future us doesn't go insane trying to figure out why metadata isn't getting included :)